src/cluster_legacy.c

```diff
@@ -3238,7 +3220,26 @@ int clusterProcessPacket(clusterLink *link) {
         }
 
         if (type == CLUSTERMSG_TYPE_MEET) {
-            if (!sender) {
+            if (!sender && link->node) {
+                /* We received a MEET packet on an existing link.
+                 * It means we received a second MEET packet from a node during the handshake
+                 * process before we were able to send a PING packet to that node from our outbound
+                 * connection.
+                 * Here we are avoiding going into the next "else if" branch so as to not assert in
+                 * setClusterNodeToInboundClusterLink() because of link->node not being NULL.
+                 *
+                 * The other sends a MEET packet because it detects that there is no inbound link,
+                 * this node creates a new node in HANDSHAKE state (with a random node name), and
+                 * respond with a PONG. The other node receives the PONG and removes the
+                 * CLUSTER_NODE_MEET flag.
+                 * This node is supposed to open an outbound connection to the other node in the
+                 * next cron cycle, but before this happens, the other node might re-send a MEET on
+                 * the same link because it still detects no inbound connection.
+                 *
+                 * Note that in getNodeFromLinkAndMsg, the node in the handshake state has a random name
+                 * and not truly "known", so we don't know the sender. */
```

Comment:
Maybe we could simplify the comments a bit?

```suggestion
                /* A second MEET packet was received on an existing link during the handshake process.
                 * This happens when the other node detects no inbound link, and re-sends a MEET packet
                 * before this node can respond with a PING. This MEET is a no-op.
                 *
                 * Note: Nodes in HANDSHAKE state are not fully "known" (random names), so the sender 
                 * remains unidentified at this point. The MEET packet might be re-sent if the inbound 
                 * connection is still unestablished by the next cron cycle.
                 */
```